test/async-hooks/test-graph.tcp.js

```diff
@@ -17,7 +17,7 @@ const server = net
 
 server.listen(common.PORT);
 
-net.connect({ port: server.address().port, host: server.address().address },
+net.connect({ port: server.address().port, host: '::1' },
```

Comment:
I'm +1 on this and +1 on fixing `adress().adress`, but this PR is about fixing `common.hasIPv6`. I would like to make just the test suit pass, and leave such improvements for other PRs.